### PR TITLE
Promote cloudbuildv2_connection and cloudbuildv2_repository to GA

### PR DESCRIPTION
**Pull request by vicpadilla**
The Cloud Build repositories API is becoming GA.
This PR promotes the cloudbuildv2 connection and repository resources to GA, along with the cloudbuild_trigger field that makes use of those resources.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_cloudbuildv2_connection` (ga)
```

```release-note:new-resource
`google_cloudbuildv2_repository` (ga)
```

```release-note:enhancement
cloudbuild: added `repository_event_config` field to `google_cloudbuild_trigger` resource (ga)
```

### CODE REVIEW

**Comment by roaks3** [comment]
Adding @slevenick who has some context here

File: mmv1/products/cloudbuildv2/product.yaml [hunk 1]
```diff
@@ -18,6 +18,9 @@ versions:
   - !ruby/object:Api::Product::Version
     name: beta
     base_url: https://cloudbuild.googleapis.com/v2/
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://cloudbuild.googleapis.com/v2/
```
Comment (slevenick) [inline mmv1/products/cloudbuildv2/product.yaml:23]: What's the difference between the beta & GA versions of the resource? Are there features that are still beta-only here?

Comment (vicpadilla) [reply mmv1/products/cloudbuildv2/product.yaml:23]: no differences at the moment. I left the beta for consistency with our cloudbuild (v1) product/resources. Should I remove beta?

**Comment by vicpadilla** [comment]
Converted PR to draft, as this PR depends on a DCL upgrade that is being reverted at https://github.com/GoogleCloudPlatform/magic-modules/pull/8289

File: mmv1/products/cloudbuildv2/product.yaml (hunk 1 above)
Comment (vicpadilla) [reply mmv1/products/cloudbuildv2/product.yaml:20]: Done. Removed beta entries.

**Review by slevenick — APPROVED** [review]

**Pull request merged**